### PR TITLE
task resource property change

### PR DESCRIPTION
**Pull request by niadak**

### CODE REVIEW

File: Tasks/DeployAzureResourceGroup/task.json [hunk 1]
```diff
@@ -100,6 +100,35 @@
             "groupName": "advanced",
```
Comment (abhinitmodi) [inline Tasks/DeployAzureResourceGroup/task.json]: Please get the json validated from a PM. The Name which appears, the help strings etc..

File: Tasks/DeployAzureResourceGroup/DtlServiceHelper.ps1 [hunk 2]
```diff
@@ -85,6 +85,26 @@ function Create-Environment
 
     $property = New-Object Microsoft.VisualStudio.Services.DevTestLabs.Model.PropertyBagData($false, $azureResourceGroup.ResourceId)
     $propertyBag.Add("PlatformId", $property)
+
+    if ($vmCreds -eq "true")
+    {
+        if([string]::IsNullOrEmpty($vmUserName) -eq $true)
+        {
+            Throw "Please specify Username"
+        }
```
Comment (abhinitmodi) [inline Tasks/DeployAzureResourceGroup/DtlServiceHelper.ps1]: Should we add this validation logic right in the beginning

File: Tasks/DeployAzureResourceGroup/task.json [hunk 3]
```diff
@@ -100,6 +100,35 @@
...
             "helpMarkDown": "Name of the SAS token parameter.",
             "visibleRule": "dscDeployment = true"
+        },
+		{
+            "name": "vmCreds",
+            "type": "boolean",
+            "label": "Virtual Machine Credentials",
```
Comment (abhinitmodi) [inline Tasks/DeployAzureResourceGroup/task.json]: label: set environment level credential?

**Comment by abhinitmodi** [comment]
Signing off with comments

File: Tasks/DeployAzureResourceGroup/DtlServiceHelper.ps1 [hunk 4]
```diff
@@ -85,6 +85,17 @@ function Create-Environment
 
     $property = New-Object Microsoft.VisualStudio.Services.DevTestLabs.Model.PropertyBagData($false, $azureResourceGroup.ResourceId)
     $propertyBag.Add("PlatformId", $property)
+
+    if ($vmCreds -eq "true")
+    {
+        $usernameTagKey = "Microsoft-Vslabs-MG-Resource-Username"
+        $property = New-Object Microsoft.VisualStudio.Services.DevTestLabs.Model.PropertyBagData($false, $vmUserName)
+        $propertyBag.Add($usernameTagKey, $property)
```
Comment (pankajkhanzode) [inline Tasks/DeployAzureResourceGroup/DtlServiceHelper.ps1]: Safe way is - AddOrUpdate, can we change to that?

File: Tasks/DeployAzureResourceGroup/DtlServiceHelper.ps1 [hunk 5]
```diff
@@ -85,6 +85,17 @@ function Create-Environment
...
+
+        $passwordTagKey = "Microsoft-Vslabs-MG-Resource-Password"
+        $property = New-Object Microsoft.VisualStudio.Services.DevTestLabs.Model.PropertyBagData($true, $vmPassword)
+        $propertyBag.Add($passwordTagKey, $property)
```
Comment (pankajkhanzode) [inline Tasks/DeployAzureResourceGroup/DtlServiceHelper.ps1]: Use AddOrUpdate

**Comment by pankajkhanzode** [comment]
Signing off with comments - please use AddOrUpdate. Then go ahead with merge.

**Comment by abhinitmodi** [comment]
Could u please send a CR iteration. It would make it easier for reviewing. Have a few minor suggestion

**Pull request merged**